File: tree.cpp
//
// Created by alisa on 23.10.2021.
//

#include "tree.h"
#include "cstring"

void Tree::In(FILE *ifst) {
    long input_age;
    if (fscanf(ifst, "%ld", &input_age) != EOF && input_age > 0) {
        age = input_age;
    }
}

void Tree::InRnd() {
    age = rand() % 100000 + 1000;
}

void Tree::Out(FILE *ofst) {
    fprintf(ofst, "It is tree: name = %s, age = %ld. Quotient = %f\n", name, age, GetQuotient());
}

double Tree::GetQuotient() {
    double result = 0;
    int length = strlen(name);
    int vowels = 0;
    for (int i = 0; i < strlen(name); i++) {
        if (name[i] == 'a' || name[i] == 'o' || name[i] == 'u' || name[i] == 'y' || name[i] == 'e' || name[i] == 'i')
            vowels++;
    }
    result = (double) vowels / length;
    return result;
}

File: tree.py
from plant import *


# ----------------------------------------------
class Tree(Plant):
    def __init__(self):
        super().__init__()
        self.age = 0

    def ReadStrArray(self, strArray, i):
        if i >= len(strArray):
            return 0
        self.age = int(strArray[i])
        i += 1
        return i

    def Print(self, i):
        print(i, ": Tree: age = ", self.age, ", name = ", self.name, ", Quotient = ", self.GetQuotient())
        pass

    def Write(self, ostream, i):
        ostream.write("{}: Tree: age = {}, name = {}, GetQuotient = {}".format \
                          (i, self.age, self.name, self.GetQuotient()))
        pass

    def GetQuotient(self):
        length = len(self.name)
        vowels = 0
        for i in range(length):
            if self.name[i] == 'a' or self.name[i] == 'o' or self.name[i] == 'e' or self.name[i] == 'u' \
                    or self.name[i] == 'i':
                vowels += 1
        return float(vowels / length)
        pass
